### PR TITLE
Upgrading sipjs to 0.10.0

### PR DESCRIPTION
**Pull request by vyshakhbabji**

### CODE REVIEW

File: src/ringcentral-web-phone.js [hunk 1]
```diff
@@ -1034,6 +995,73 @@
...
+    /**
+     * @this {SIP.Session}
+     * @return {Promise}
+     */
+
+    function reinvite (options, modifier){
+        var session = this;
+        options = options || {}
+        options.sessionDescriptionHandlerOptions = options.sessionDescriptionHandlerOptions || {};
+        options.sessionDescriptionHandlerOptions.constraints = options.sessionDescriptionHandlerOptions.constraints || {audio: true, video: false};
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: Can we add this to defaults and call the original `reinvite` directly?

File: src/ringcentral-web-phone.js [hunk 2]
```diff
@@ -836,74 +864,7 @@
...
-                            session.removeListener('RC_SIP_NOTIFY', onNotify);
-                        }, responseTimeout);
-                        session.on('RC_SIP_NOTIFY', onNotify);
-                    }
-                    else {
-                        reject(new Error('The response status code is: ' + response.status_code + ' (waiting for 202)'));
-                    }
-                }
-            });
-
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: RC_SIP_NOTIFY can now be removed completely.

File: src/ringcentral-web-phone.js [hunk 3]
```diff
@@ -369,8 +394,8 @@
             session.removeListener('cancel', stopPlaying);
             session.removeListener('failed', stopPlaying);
             session.removeListener('replaced', stopPlaying);
-            session.mediaHandler.removeListener('iceConnectionCompleted', stopPlaying);
-            session.mediaHandler.removeListener('iceConnectionFailed', stopPlaying);
+            // session.sessionDescriptionHandler.removeListener('iceConnectionCompleted', stopPlaying);
+            // session.sessionDescriptionHandler.removeListener('iceConnectionFailed', stopPlaying);
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: Remove if it's not needed, don't leave it commented

File: src/ringcentral-web-phone.js [hunk 4]
```diff
@@ -356,8 +381,8 @@
         session.on('cancel', stopPlaying);
         session.on('failed', stopPlaying);
         session.on('replaced', stopPlaying);
-        session.mediaHandler.on('iceConnectionCompleted', stopPlaying);
-        session.mediaHandler.on('iceConnectionFailed', stopPlaying);
+        // session.sessionDescriptionHandler.on('iceConnectionCompleted', stopPlaying);
+        // session.sessionDescriptionHandler.on('iceConnectionFailed', stopPlaying);
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: Remove if it's not needed, don't leave it commented

File: src/ringcentral-web-phone.js [hunk 5]
```diff
@@ -1034,6 +995,73 @@
...
+    };
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    /**
+     * @this {SIP.Session}
+     * @return boolean
+     */
+
+    function onHold (){
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: Rename to setLocalHold

Comment (vyshakhbabji) [reply src/ringcentral-web-phone.js]: Renamed setHold function to setLocalHold

File: src/ringcentral-web-phone.js (hunk 3 above)
Comment (vyshakhbabji) [reply src/ringcentral-web-phone.js]: Removed

File: src/ringcentral-web-phone.js (hunk 4 above)
Comment (vyshakhbabji) [reply src/ringcentral-web-phone.js]: Removed

File: src/ringcentral-web-phone.js (hunk 2 above)
Comment (vyshakhbabji) [reply src/ringcentral-web-phone.js]: Removed

File: src/ringcentral-web-phone.js (hunk 3 above)
Comment (kirill-konshin) [reply src/ringcentral-web-phone.js]: I have asked to remove! Not uncomment!

File: src/ringcentral-web-phone.js (hunk 4 above)
Comment (kirill-konshin) [reply src/ringcentral-web-phone.js]: I have asked to remove! Not uncomment!

File: demo/index.js [hunk 6]
```diff
@@ -220,11 +221,36 @@ $(function() {
...
+            // Gets local tracks
+            var localVideo = document.getElementById('localVideo');
+            var localStream = new MediaStream();
+            pc.getSenders().forEach(function(sender) {
+                localStream.addTrack(sender.track);
+            });
+            localVideo.srcObject = localStream;
+            localVideo.play();
+    }
+
```
Comment (kirill-konshin) [inline demo/index.js]: Remove this from demo completely, put this inside the SDK as a part of session patching procedure. SDK constructor should receive two references to `localVideo` and `remoteVideo`.

File: demo/index.js [hunk 7]
```diff
@@ -220,11 +221,36 @@ $(function() {
...
     function onAccepted(session) {
 
         console.log('EVENT: Accepted', session.request);
         console.log('To', session.request.to.displayName, session.request.to.friendlyName);
         console.log('From', session.request.from.displayName, session.request.from.friendlyName);
+        session.on('trackAdded',function () {
+            addTrack(session);
+        });
```
Comment (kirill-konshin) [inline demo/index.js]: Remove this too.

Comment (kirill-konshin) [reply demo/index.js]: Also remove occurrences of

```
var acceptOptions = {
             media: {
                 render: {
                     remote: document.getElementById('remoteVideo'),
                     local: document.getElementById('localVideo')
                 }
             }
         };
```

File: src/ringcentral-web-phone.js [hunk 8]
```diff
@@ -1035,6 +982,115 @@
...
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+
+    function addTrack(){
+
+        var session = this;
+        var pc = session.sessionDescriptionHandler.peerConnection;
+
+        // Gets remote tracks
+        var remoteAudio = media.remote;
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: This should be part of SESSION

File: src/ringcentral-web-phone.js [hunk 9]
```diff
@@ -1035,6 +982,115 @@
...
+                }});
+        }
+        else{
+            remoteStream = pc.getRemoteStreams()[0];
+        }
+        remoteAudio.srcObject = remoteStream;
+        remoteAudio.play();
+
+        // Gets local tracks
+        var localAudio = media.local;
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: This should be part of SESSION

**Pull request merged**